### DISCUSSION TITLE
How is it better than PNPM?

### DISCUSSION THREAD

**Discussion by babakfp** [denoland/deno#16618] [Q&A] [answered]
Hey
I live in Iran and right now it's not possible to run a Deno project because of this https://github.com/denoland/dotland/issues/2161
I use PNPM for my projects and because of its awesome features, I don't need to download a package again if I have already downloaded it! This is a great feature and it makes me and others continue working on their projects instead of not doing anything or trying to install the package again whenever the internet was available.
Now my question is, do I need to download packages again and again with Deno? Or how it works? How is it better than PNPM? Is it something that just sounds cool (you don't need node_modules)?

**Comment by dsherret** [comment, marked as answer]
> I use PNPM for my projects and because of its awesome features, I don't need to download a package again if I have already downloaded it! ... Now my question is, do I need to download packages again and again with Deno?

No, once a module or npm package is downloaded to the global cache it won't be downloaded again unless you use the `--reload` flag.
